packages/api/src/pins.js

```diff
@@ -56,11 +56,8 @@ export const getPinningAPIStatus = (pins) => {
     return 'queued'
   }
 
-  if (pinStatuses.length === 0) {
-    return 'queued'
-    // TODO after some time if there are no pins we should give up and return a failed
-    // status instead
-  }
+  // TODO after some time if there are no pins we should give up and return a failed
+  // status instead
```

Comment:
Seems that is what we do now, straight away. Maybe this can be removed?

Reply:
I guess this relates to the fact that it'd be nice to have some kind of background task to check after a given defined threshold (an hour, a day, a week I don't know) to tell the cluster to stop trying to pin the cid (I know you have priority queues now but I still feel it should be good to programmatically stop the cluster trying).
At the same time, we should update the DB and therefore the request status accordingly.

I don't think this is a super high priority but still worth tracking?
If you agree I can update the comment to be more verbose and tie it with a github issue.

Reply:
ping 😇